src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,155 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidDataException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static final int MIN_AGE = 18;
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    void setup() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_validUser_successfulRegistration() {
+        User user = new User();
+        user.setLogin("username");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser);
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+    }
+
+    @Test
+    public void register_existingUser_throwException() {
+        User user1 = new User();
+        user1.setLogin("username");
+        user1.setPassword("password123");
+        user1.setAge(20);
+
+        User user2 = new User();
+        user2.setLogin("username");
+        user2.setPassword("anotherpassword");
+        user2.setAge(25);
+
+        Exception exception = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+            registrationService.register(user2);
+        });
+
+        String expectedMessage = "User with this login already exists";
+        String actualMessage = exception.getMessage();
+
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    public void register_shortLogin_throwException() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        Exception exception = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user));
+
+        String expectedMessage = "Login and password must be at least 6 characters long";
+        String actualMessage = exception.getMessage();
+
+        assertTrue(actualMessage.contains(expectedMessage));
```

Comment:
`assertEquals("message", exception.getMessage())`

**apply to all places where necessary**